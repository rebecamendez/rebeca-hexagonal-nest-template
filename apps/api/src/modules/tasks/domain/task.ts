export class Task {
  public id!: number;
  public title!: string;
  public description!: string;
}
