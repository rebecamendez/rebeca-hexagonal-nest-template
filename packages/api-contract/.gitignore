# Dependencies
node_modules
.pnp
.pnp.js

# Build outputs
dist
build

# Testing
coverage

# IDE
.idea
.vscode
*.swp
*.swo

# Logs
logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*
.pnpm-debug.log*

# Environment
.env
.env.local
.env.development.local
.env.test.local
.env.production.local

# TypeScript
*.tsbuildinfo 