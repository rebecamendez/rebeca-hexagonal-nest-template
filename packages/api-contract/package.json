{
  "name": "@rebeca-hexagonal-nest-template/api-contract",
  "version": "0.1.0",
  "private": true,
  "author": "Rebeca MÃ©ndez",
  "module": "dist/esm/index.js",
  "main": "dist/cjs/index.js",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsc -p tsconfig.browser.json && tsc -p tsconfig.node.json && node scripts/build-swagger-api-contract.js",
    "build:swagger": "node scripts/build-swagger-api-contract.js",
    "clean": "rm -rf node_modules .turbo dist coverage tsconfig.tsbuildinfo",
    "deps:update": "pnpm update -r",
    "dev": "tsc --watch",
    "format": "prettier --write \"src/**/*.ts\" --config ./prettier.config.js",
    "lint": "eslint \"src/**/*.ts\" --config ./eslint.config.js"
  },
  "exports": {
    ".": "./dist/cjs/index.js",
    "./openapi": "./dist/openapi.json"
  },
  "dependencies": {
    "class-transformer": "0.5.1",
    "class-validator": "0.14.1"
  },
  "devDependencies": {
    "@rebeca-hexagonal-nest-template/lint-config": "workspace:*",
    "@types/node": "22.13.13",
    "rimraf": "6.0.1",
    "typescript": "5.8.2"
  }
}
