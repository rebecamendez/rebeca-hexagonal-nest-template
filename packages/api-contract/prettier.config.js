const prettierConfig = require("@rebeca-hexagonal-nest-template/lint-config/prettier.js");

module.exports = prettierConfig; 
