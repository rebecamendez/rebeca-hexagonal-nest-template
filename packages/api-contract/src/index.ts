export * from './tasks/task.request';
export * from './tasks/task.request.mock';
export * from './tasks/task.response';
export * from './tasks/task.response.mock';
