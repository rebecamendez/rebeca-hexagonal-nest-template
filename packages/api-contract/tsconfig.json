{
  "compilerOptions": {
    "target": "ES2020",
    "sourceMap": true,
    "composite": true,
    "rootDir": "./src",
    "outDir": "./dist",
    "module": "Node16",
    "moduleResolution": "node16",
    "strict": true,
    "esModuleInterop": true,
    "skipLibCheck": true,
    "forceConsistentCasingInFileNames": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "declarationMap": true,
    "noImplicitAny": true,
    "noUnusedLocals": true,
    "removeComments": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "noImplicitThis": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "allowSyntheticDefaultImports": true,
    "resolveJsonModule": true,
    "alwaysStrict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "exactOptionalPropertyTypes": true,
    "allowUnreachableCode": false,
    "allowUnusedLabels": false,
    "strictPropertyInitialization": true,
    "useUnknownInCatchVariables": true,
    "noUncheckedIndexedAccess": true,
    "noImplicitUseStrict": false,
    "declaration": true
  },
  "exclude": [
    "node_modules"
  ],
  "include": [
    "./src/**/*.ts"
  ]
}
