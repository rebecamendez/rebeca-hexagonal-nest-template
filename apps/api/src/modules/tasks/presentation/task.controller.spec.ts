import { createTaskRequestMock } from '@rebeca-hexagonal-nest-template/api-contract';
import { mock, mockReset } from 'jest-mock-extended';

import { TaskService } from '../application/task.service';
import { createTaskMock } from '../domain/task.mock';
import { TaskController } from './task.controller';

describe('A Task controller', () => {
  const service = mock<TaskService>();
  const controller = new TaskController(service);

  beforeEach(() => {
    mockReset(service);
  });

  it('should get all tasks', async () => {
    const tasks = [createTaskMock()];

    service.getTasks.calledWith().mockResolvedValue(tasks);

    const result = await controller.getTasks();
    expect(result).toMatchSnapshot();
  });

  it('should get a task by id', async () => {
    const task = createTaskMock();

    service.getTask.calledWith(task.id).mockResolvedValue(task);

    const result = await controller.getTask(task.id);
    expect(result).toMatchSnapshot();
  });

  it('should create a task', async () => {
    const request = createTaskRequestMock();
    const task = createTaskMock();

    service.createTask.calledWith(request.title, request.description).mockResolvedValue(task);

    const result = await controller.createTask(request);
    expect(result).toMatchSnapshot();
  });

  it('should update a task', async () => {
    const request = createTaskRequestMock();
    const task = createTaskMock();

    service.updateTask.calledWith(task.id, request.title, request.description).mockResolvedValue(task);

    const result = await controller.updateTask(task.id, request);
    expect(result).toMatchSnapshot();
  });

  it('should delete a task', async () => {
    const task = createTaskMock();

    await controller.deleteTask(task.id);
    expect(service.deleteTask).toHaveBeenCalledWith(task.id);
  });
});
