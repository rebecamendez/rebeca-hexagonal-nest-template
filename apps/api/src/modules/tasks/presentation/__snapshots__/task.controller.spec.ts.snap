// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Task controller should create a task 1`] = `
TaskResponse {
  "description": "Test Description",
  "id": 1,
  "title": "Test Task",
}
`;

exports[`A Task controller should get a task by id 1`] = `
TaskResponse {
  "description": "Test Description",
  "id": 1,
  "title": "Test Task",
}
`;

exports[`A Task controller should get all tasks 1`] = `
[
  TaskResponse {
    "description": "Test Description",
    "id": 1,
    "title": "Test Task",
  },
]
`;

exports[`A Task controller should update a task 1`] = `
TaskResponse {
  "description": "Test Description",
  "id": 1,
  "title": "Test Task",
}
`;
